extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -536,28 +556,68 @@ export class AmpStoryEmbeddedComponent {
-          const centeredLeft = pageRect.width / 2 - targetRect.width / 2;
-
-          // Only account for offset from target to page borders. Since in
-          // desktop mode page is not at the borders of viewport.
-          const leftOffset = targetRect.left - pageRect.left;
-          const topOffset = targetRect.top - pageRect.top;
-
-          state.translateY = centeredTop - topOffset;
-          state.translateX = leftOffset - centeredLeft;
+          this.scaleFactor_ =
+            parseFloat(getStyle(target,'transform').split('scale(')[1]);
+          const realWidth = targetRect.width / this.scaleFactor_;
+          const leftOffset = (realWidth - targetRect.width) / 2;
+          const horizontalOffset = targetRect.left - leftOffset - pageRect.left;
+          const centeredLeft = pageRect.width / 2 - realWidth / 2;
+          state.translateX = centeredLeft - horizontalOffset;
+
+          const realHeight = targetRect.height / this.scaleFactor_;
+          const topOffset = (realHeight - targetRect.height) / 2;
+          const verticalOffset = targetRect.top - topOffset - pageRect.top;
+          const centeredTop = pageRect.height / 2 - realHeight / 2;
+          state.translateY = centeredTop - verticalOffset;
         },
         /** mutate */
         () => {
           target.classList.add('i-amphtml-animate-expand-in');
+          target.classList.toggle('i-amphtml-expanded-component', true);
           setImportantStyles(dev().assertElement(target),
               {
                 transform: `translate3d(${state.translateX}px,
-                    ${state.translateY}px, 0)`,
+                  ${state.translateY}px, 0) scale(1)`,
               });
         });
   }
 
+  /**
+   * Prepares expandable component for its expanded animation.
+   * It resizes it to its full-screen size, and scales it down to match size set
+   * by publisher, adding negative margins so that content around stays put.
+   * @param {!Element} page
+   * @param {!Element} element
+   */
+  static prepareForAnimation(page, element) {
+    element.whenBuilt().then(() => {
+      element.classList.add('i-amphtml-embedded-component');
+      const pageRect = page./*OK*/getBoundingClientRect();
+      const elRect = element./*OK*/getBoundingClientRect();
+
+      let scaleFactor, newHeight, newWidth;
+      if (elRect.width >= elRect.height) {
+        newWidth = pageRect.width;
+        scaleFactor = elRect.width / newWidth;
+        newHeight = elRect.height / elRect.width * newWidth;
+      } else {
+        newHeight = pageRect.height - VERTICAL_BUFFER;
+        scaleFactor = elRect.height / newHeight;
+        newWidth = elRect.width / elRect.height * newHeight;
+      }
+
+      const verticalMargin = (-1 * ((newHeight - elRect.height) / 2));
+      const horizontalMargin = (-1 * ((newWidth - elRect.width) / 2));
+
+      setImportantStyles(devAssert(element), {
+        width: `${newWidth}px`,
```

Comment:
nit: use `px()` from `src/style.js` here and below

Reply:
Done.